nav2_navfn_planner/src/navfn_planner.cpp

```diff
@@ -321,7 +321,13 @@ NavfnPlanner::getPlanFromPotential(
 
   planner_->setStart(map_goal);
 
-  planner_->calcPath(costmap_->getSizeInCellsX() * 4);
+  int path_len = planner_->calcPath(costmap_->getSizeInCellsX() * 4);
```

Comment:
Do you know what's the significance of this number? Why size_x*4?

Reply:
Its a psuedo-magic number. Its meant to bound the total amount of expansion so that in the case of no valid plan its not going over the top and searching the while map may times over.

Reply:
@crdelsey I am afraid I don't quite sure why. It is still a mystery for me. It is representing the total number of cell down the code but don't know why it is `size_x*4`.